File: hackerrank/30 days of code/day 6, lets review/lets_review.py
# https://www.hackerrank.com/challenges/30-review-loop/problem?isFullScreen=true

def main():
    n = int(input())

    for _ in range(n):
        word = input()
        word1, word2 = '', ''
        is_word1_turn = True

        for ch in word.strip():
            if is_word1_turn:
                word1 += ch
            else:
                word2 += ch
            is_word1_turn = not is_word1_turn

        print(f"{word1} {word2}")


if __name__ == '__main__':
    main()

File: hackerrank/30 days of code/day 6, lets review/lets_review.go
// https://www.hackerrank.com/challenges/30-review-loop/problem?isFullScreen=true
// From Go 1.22 onwards

package main

import "fmt"

func main() {
    var n int
    fmt.Scan(&n)

    for range n {
        var word string
        fmt.Scan(&word)
        var word1, word2 string
        isWord1Turn := true

        for _, ch := range word {
            if isWord1Turn {
                word1 += string(ch)
            } else {
                word2 += string(ch)
            }
            isWord1Turn = !isWord1Turn
        }

        fmt.Println(word1, word2)
    }
}
